todo.c

```diff
@@ -45,21 +45,27 @@ void todo_swap(todo_t *a, todo_t *b) {
 }
 
 void todo_add_sort_option(enum sort_option opt) {
-	//sort_options_count++;
 	sort_options = realloc(sort_options, sizeof(int) * (++sort_options_count));
 	sort_options[sort_options_count - 1] = opt;
 }
 
+
+/*
+	I use bubble sort because it is one of the simplest,
+	even if it is very slow. Because, i could add custom
+	ordering rules when i know how the algorithm works.
+*/
```

Comment:
For your application even qsort could be simple. You would write comparing function which would have the decision login in it (not just many simple comparison functions, but one with flags so it knew how to compare them).

Reply:
I would like to implement `qsort` in the future. Seems potential if someone have much things in their ToDo List